components/bsx/Gallery/Item/DangerModal.vue

```diff
@@ -0,0 +1,89 @@
+<template>
+  <ModalWrapper icon="trash" :title="title" isRight>
+    <template v-slot:default>
+      <Loader v-model="isLoading" :status="status" />
+      <div class="buttons">
+        <b-button type="is-danger" expanded outlined @click="clearMeta"
+          >Remove metadata</b-button
+        >
+        <b-button
+          type="is-danger"
+          expanded
+          outlined
+          @click="clearAttrs"
+          :disabled="!hasAttributes"
+          >Remove ALL attributes</b-button
+        >
+      </div>
+    </template>
+  </ModalWrapper>
+</template>
+
+<script lang="ts">
+import { Component, mixins, Prop, Watch } from 'nuxt-property-decorator'
+import Connector from '@kodadot1/sub-api'
+import MetaTransactionMixin from '@/utils/mixins/metaMixin'
+import { showNotification } from '~/utils/notification'
+import { Attribute } from '../../types'
+import { emptyArray } from '@/utils/empty'
+
+const components = {
+  Loader: () => import('@/components/shared/Loader.vue'),
+  ModalWrapper: () => import('@/components/shared/modals/ModalWrapper.vue'),
+}
+
+@Component({
+  name: 'DangerModal',
+  components,
+})
+export default class DangerModal extends mixins(MetaTransactionMixin) {
+  @Prop({ type: String }) public accountId!: string
+  @Prop({ type: String }) public title!: string
+  @Prop({ type: String, required: true }) public collectionId!: string
+  @Prop(String) public nftId!: string
+  @Prop({ type: Array, default: () => emptyArray<Attribute>() })
+  public attributes!: Attribute[]
+
+  protected clearMeta() {
+    this.submit(true)
+  }
+
+  protected clearAttrs() {
+    this.submit(false)
+  }
+
+  get hasAttributes() {
+    return this.attributes && this.attributes.length
+  }
+
+  protected async submit(isMeta: boolean) {
+    const { api } = Connector.getInstance()
+    this.initTransactionLoader()
+    const cb = isMeta ? api.tx.uniques.clearMetadata : api.tx.utility.batchAll
+    const args = isMeta
+      ? [this.collectionId, this.nftId]
+      : [
+          this.attributes
+            .map((attr) => attr.key)
+            .map((key) =>
+              api.tx.uniques.clearAttribute(this.collectionId, this.nftId, key)
+            ),
+        ]
+
+    this.howAboutToExecute(this.accountId, cb, args, this.onSuccess)
+
+    this.$emit('close')
+  }
```

Comment:
I personally would prefer the following way, each func doing one thing. This may seem verbose firstly, but the reader does not need to do a logic jump, which reduces complexitiy.

```js
protected clearMeta() {
    const { api } = Connector.getInstance()
    this.submit(api.tx.uniques.clearMetadata, [this.collectionId, this.nftId])
}

protected clearAttrs() {
    const { api } = Connector.getInstance()
    const args = [
        this.attributes
            .map((attr) => attr.key)
            .map((key) =>
                api.tx.uniques.clearAttribute(this.collectionId, this.nftId, key)
            ),
    ]
    this.submit(api.tx.utility.batchAll, args)
}

protected async submit(cb, args) {
    this.initTransactionLoader()
    this.howAboutToExecute(this.accountId, cb, args, this.onSuccess)
    this.$emit('close')
}
```

Reply:
Like this approach! :)